deepmd/tf/fit/polar.py

```diff
@@ -193,18 +193,16 @@ def compute_output_stats(self, all_stat):
                         index_lis = [
                             index
                             for index, w in enumerate(atom_has_polar)
-                            if atom_has_polar[index] == self.sel_type[itype]
+                            if w == self.sel_type[itype]
                         ]  # select index in this type
 
                         sys_matrix.append(np.zeros((1, len(self.sel_type))))
                         sys_matrix[-1][0, itype] = len(index_lis)
 
                         polar_bias.append(
                             np.sum(
-                                all_stat["atomic_polarizability"][ss].reshape((-1, 9))[
-                                    index_lis
-                                ],
-                                axis=0,
+                                all_stat["atomic_polarizability"][ss][:, index_lis, :],
+                                axis=(0, 1),
```

Comment:
currently, the `TF` only considers the first frame when doing aggregation. This need to consider all frames.

Also, there is a factor of `nframe` differences between the `TF` and `NP` implementation.